site/components/pages/Pledge/lib/verifyGnosisSafeMultisig.ts

```diff
@@ -0,0 +1,32 @@
+import { ethers } from "ethers";
+import { getJsonRpcProvider } from "@klimadao/lib/utils";
+import GnosisSafeSignMessageLib from "@klimadao/lib/abi/GnosisSafeSignMessageLib.json";
+
+const ONE_HOUR = 60 * 60 * 1000;
+
+export const verifyGnosisSafeMultisig = async (params: {
+  signature: string;
+  address: string;
+}): Promise<void> => {
+  const provider = getJsonRpcProvider();
+  const gnosisSafeContract = new ethers.Contract(
+    params.address,
+    GnosisSafeSignMessageLib.abi,
+    provider
+  );
+  const messageHash = ethers.utils.hashMessage(params.signature);
+  const getMessageHash = await gnosisSafeContract.getMessageHash(messageHash);
+  const signedEvent = gnosisSafeContract.filters.SignMsg(getMessageHash);
+
+  const waitForSignedEvent = (): Promise<void> =>
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        gnosisSafeContract.removeListener(signedEvent, () => resolve());
+        reject(new Error("Gnosis safe signature verification timed out"));
+      }, ONE_HOUR);
+
+      gnosisSafeContract.once(signedEvent, () => resolve());
```

Comment:
does this only succeed on future events? so we aren't able to do a flow like this:
1. wait clientside for event
2. pass signature to server
3. look back at recent events and see if it exists

?